final-project/application.py

```diff
@@ -563,28 +565,139 @@ def parse_posted_json():
 
 
 # PART 2: Update the database.
-# TODO:
-def update_database(the_new_info):
+def update_database(requested_product_updates):
+
+    print('')
+    print('Request to update the database was heard')
+    print('')
+    print('here are the requested updates')
+    print(requested_product_updates)
+    print('')
 
     # Update only the information our user asked to change.
-    if not the_new_info['product-id']:
+    if not requested_product_updates['product-id']:
         return apology("Unsure which product to update")
-    if 'name-form' in the_new_info:
-        updated_row = db.execute("""
-            UPDATE products SET product_name=:name""",
-            name=the_new_info['name-form'])
-    if 'brand-form' in the_new_info:
-        updated_row = db.execute("""
-            UPDATE products SET brand=:brand""",
-            brand=the_new_info['brand-form'])
-    if 'price-form' in the_new_info:
-        updated_row = db.execute("""
-            UPDATE products SET price=:price""",
-            price=the_new_info['price-form'])
-    if 'description-form' in the_new_info:
-        updated_row = db.execute("""
-            UPDATE products SET description=:description""",
-            description=the_new_info['description-form'])
+    if 'name-form' in requested_product_updates:
+        number_of_updates = db.execute("""
+            UPDATE products SET product_name=:name"""
+            , name=requested_product_updates['name-form'])
+    if 'brand-form' in requested_product_updates:
+        number_of_updates = db.execute("""
+            UPDATE products SET brand=:brand"""
+            , brand=requested_product_updates['brand-form'])
+    if 'price-form' in requested_product_updates:
+        number_of_updates = db.execute("""
+            UPDATE products SET price=:price"""
+            , price=requested_product_updates['price-form'])
+    if 'description-form' in requested_product_updates:
+        number_of_updates = db.execute("""
+            UPDATE products SET description=:description"""
+            , description=requested_product_updates['description-form'])
+
+    # # TODO: Check for references in requested_product_updates
+    if 'reference-form' in requested_product_updates:
```

Comment:
This whole block seems incredibly complex. Think you could break it up?

Reply:
Yeah, it's terrible. I started to break all the functionality out into separate functions, but it was taking too long to figure out how to group everything. I am sure if I followed the functional programming principles you recommended, then this would never be an issue. But, I am just trying to wrap this up so I scrapped the idea to break this up.

I have all the Read, Update, and Delete functionality done for products.

The only thing left is Create. I have to give users a way to create a reference. 

When that create functionality is done, then I will clean up the code and submit to CS50. So close I can taste it.

Reply:
@hdavidzhu

Reply:
Haha just something to keep in mind for future programming. Let's get to the finish line